fs/fs.go

```diff
@@ -409,7 +409,7 @@ func (fs *filesystem) Mount(ctx context.Context, mountpoint string, labels map[s
 	client := src[0].Hosts[0].Client
 	c, err := fs.getSociContext(ctx, imageRef, sociIndexDigest, imgDigest, client)
```

Comment:
I suppose this could be it's own issue/discussion, but right now we fallback either to overlayfs/the default snapshotter even on partial artifact fetch success, meaning we were able to fetch the index and only _some_ zTOC's. Should that be the case? Should we still try to remote mount layers that correspond to zTOC's that we _were_ able to fetch?

Reply:
To be clear, this is when the index specifies a number of zTOCs, but we get less than that expected number, correct? What scenarios might cause this to happen?

I think this behavior is probably fine since I mostly see it as a network issue if this fails (e.g. auth revoked, internet conenction lost, etc), and so it's somewhat reasonable to assume that the user no longer has access to the required repo at all.

Reply:
> To be clear, this is when the index specifies a number of zTOCs, but we get less than that expected number, correct?

Yeah.

>  What scenarios might cause this to happen?

Not exactly sure, but I'm imagining network failures.